### PR TITLE
Allow HyperLogLog<T> where T: !Sized

### PR DESCRIPTION
**Pull request by mbrubeck**
This is useful for cases like `HyperLogLog<[u8]>` where the `[u8]` slice doesn't outlive the HyperLogLog (so you can't use `HyperLogLog<&[u8]>`).

### CODE REVIEW

**Comment by mbrubeck** [comment]
example usage:

```rust
let mut h: HyperLogLog<[u8]> = HyperLogLog::new(4);
{
    let v = vec![0; 64];
    h.add(&v);
}
// `h` is still usable after `v` is dropped:
let n = h.count();
```

**Comment by crepererum** [comment]
Ah, sorry for the extremely late review. This PR kinda slipped through my notifications.

Can you please add a test for the vec case and the extend implementation?

**Comment by mbrubeck** [comment]
Added tests.

File: src/hyperloglog.rs [hunk 1]
```diff
@@ -551,4 +562,19 @@ mod tests {
...
+    #[test]
+    fn slice() {
+        // `HyperLogLog<T>` with a dynamically-sized `T`.
+        let mut h: HyperLogLog<[u8]> = HyperLogLog::new(4);
+        {
+            // Can `add` and `extend` from vecs or arrays.
+            let v1 = vec![0; 64];
+            h.add(&v1);
+            let v2 = vec![&v1[..]];
+            h.extend(v2);
```
Comment (crepererum) [inline src/hyperloglog.rs]: can you please pull out the extend part into a separate test?

File: src/hyperloglog.rs [hunk 2]
```diff
@@ -551,4 +562,19 @@ mod tests {
...
+        }
+        // `h` is still usable after `v1` is dropped:
+        h.count();
```
Comment (crepererum) [inline src/hyperloglog.rs]: please assert the count here to double-check the add above really works (should probably be `1` at this point)

File: src/hyperloglog.rs (hunk 1 above)
Comment (mbrubeck) [reply src/hyperloglog.rs]: Done.

File: src/hyperloglog.rs (hunk 2 above)
Comment (mbrubeck) [reply src/hyperloglog.rs]: Done.

**Comment by crepererum** [comment]
Thanks a lot :)

**Pull request merged**